src/ranking/ranking.ts

```diff
@@ -85,35 +88,50 @@ export function getScore(model: SpecQueryModel, rankingName: string, schema: Sch
   return score;
 }
 
-export function comparatorFactory(name: string | string[], schema: Schema, opt: QueryConfig) {
+export function comparatorFactory(name: string | string[] | SortByFieldOrder | SortByFieldOrder[], schema: Schema, opt: QueryConfig) {
   return (m1: SpecQueryModel, m2: SpecQueryModel) => {
     if (name instanceof Array) {
       return getScoreDifference(name, m1, m2, schema, opt);
-    } else {
+    } else if (typeof name === 'string') {
       return getScore(m2, name, schema, opt).score - getScore(m1, name, schema, opt).score;
+    } else {
+      return getScore(m2, name.fn, schema, opt).score - getScore(m1, name.fn, schema, opt).score;
     }
   };
 }
 
-export function groupComparatorFactory(name: string | string[], schema: Schema, opt: QueryConfig) {
+export function groupComparatorFactory(name: string | string[] | SortByFieldOrder | SortByFieldOrder[], schema: Schema, opt: QueryConfig) {
   return (g1: SpecQueryModelGroup, g2: SpecQueryModelGroup) => {
     const m1 = g1.getTopSpecQueryModel();
     const m2 = g2.getTopSpecQueryModel();
     if (name instanceof Array) {
       return getScoreDifference(name, m1, m2, schema, opt);
-    } else {
+    } else if (typeof name === 'string') {
```

Comment:
I thought we have exhaustively discuss about duplication of `getScoreDifference` here?